muda/deformers/filter.py

```diff
@@ -0,0 +1,538 @@
+        if sigma <= 0:
+            raise ValueError("sigma must be strictly positive")
+
+        if n_samples <= 0:
+            raise ValueError("n_samples must be None or positive")
+
+        if order <= 0 or attenuation <= 0:
+            raise ValueError("order and attenuation must be None or positive")
+
+        if type(cutoff)==list or cutoff<=0:
+            raise ValueError("high pass cutoff frequency must be strictly positive and lower than nyquist frequency")
+
+
+        self.n_samples = n_samples
+        self.order = order
+        self.attenuation = attenuation
+        self.sigma = float(sigma)
+        self.rng = rng
+        self._rng = _get_rng(rng)
+        self.cutoff = float(cutoff)
+      
+
+
+    #specify and stores the type/parameters of the augmentation
+
+    def states(self, jam):
+        for state in AbstractFilter.states(self, jam):
+            for _ in range(self.n_samples):
+                state["btype"] = "high"
+                state["order"] = self.order
+                state["attenuation"] = self.attenuation
+                state["cut_off"] = self._rng.normal(
+                        loc=self.cutoff, scale=self.sigma, size=None
+                    )
+                
+                yield state
+
+
+class RandomBPFilter(AbstractFilter):
+    """ Filtering by cheby2 iir filter
+
+    This transformation affects the following attributes:
+
+    - Annotations 
+        - pitch_contour, pitch_hz, pitch_midi, pitch_class
+        - note_hz, note_midi
+    - Audio
+
+    Attributes
+    ----------
+    n_samples : int > 0
+        The number of samples to generate per input
+
+    order: int > 0
+        order of the filter
+    attenuation: float > 0
+        The minimum attenuation required in the stop band. 
+        Specified in decibels, as a positive number.
+    
+    cutoff: float in hz
+        high pass cutoff frequency
+        
+    
+    
+    sigma : float > 0
+        The parameters of the normal distribution for sampling
+        pitch shifts
+
+    rng : None, int, or np.random.RandomState
+        The random number generator state.
+
+        If `None`, then `np.random` is used.
+
+        If `int`, then `rng` becomes the seed for the random state.
+
+
+    See Also
+    --------
+    Filter, RandomHPFilter, RandomLPFilter
+
+    Examples
+    --------
+    >>> # Shift down by a quarter-tone
+    >>> D = muda.deformers.RandomBPFilter(n_samples,order,attenuation,cutoff_low,cutoff_high,sigma)
+    """
+
+    def __init__(self, n_samples=3, order=4, attenuation=60.0, cutoff_low=4000, cutoff_high=8000,sigma=1.0,rng=0):
+        AbstractFilter.__init__(self)
+        if sigma <= 0:
+            raise ValueError("sigma must be strictly positive")
+
+        if n_samples <= 0:
+            raise ValueError("n_samples must be None or positive")
+
+        if order <= 0 or attenuation <= 0:
+            raise ValueError("order and attenuation must be None or positive")
```

Comment:
if `None` is a valid value, checking `order <= 0` will raise a type error.  You need to handle Nones separately:

```python
if order is not None and order <= 0:
    ...
```
It's also not good style to combine two tests and then raise a single exception, because a user then has to figure out which of the two tests failed.  Better to factor these out into separate conditional statements.